althea_kernel_interface/src/manual_peers_route.rs

```diff
@@ -0,0 +1,75 @@
+use super::{KernelInterface, KernelInterfaceError};
+
+use std::net::IpAddr;
+
+use failure::Error;
+
+impl KernelInterface {
+    pub fn manual_peers_route(
+        &self,
+        endpoint_ip: &IpAddr,
+        settings_default_route: &mut Vec<String>
+    ) -> Result<(), Error> {
+        // get default route (actually get all the routes)
+        let output = self.run_command(
+            "ip",
+            &[
+                "route",
+                "list",
+                "default",
+            ],
+        )?;
+
+        let stdout = String::from_utf8(output.stdout)?;
+
+        let endpoint_ip = format!("{}", endpoint_ip);
+
+        let mut def_route = Vec::new();
+
+        let mut found = false;
+
+        // find all lines
+        for i in stdout.split("\n") {
+            // starting with default
+            if i.starts_with("default") {
+                for j in i.split(" ") {
+                    if j.len() != 0 {
+                        def_route.push(String::from(j));
+                    }
+                }
+                found = true;
+                break;
+            }
+        }
+
+        if !found {
+            // early returning if no default route found, (e.g. on client)
+            return Ok(())
+        }
+
+        let mut default_is_wg_exit = def_route.contains(&String::from("wg_exit"));
+
+        if !default_is_wg_exit {
+            // update the default route if default route is not wg exit
+            *settings_default_route = def_route.clone();
```

Comment:
Are we editing the settings here? If so, why?

Reply:
to preserve the default route, as we will overwrite it with a default route with wg exit later